netlify.toml

```diff
@@ -1,12 +1,233 @@
 # Settings in the [build] context are global and are applied to all contexts unless otherwise overridden by more specific contexts.
 [build]
-  command = './netlify/build.sh'
+  publish = "_site/"
+  command = '''
+
+  cp netlify/Gemfile Gemfile && bundle install
+  make bin/helm
+  export PATH=$PATH:$(pwd)/bin
+
+  # LATEST_RELEASE is set in Netlify UI for environment-variables ONLY for docs.projectcalico.org site.
+  # Any branch deployed there will have the production ready build that gets deployed at root /
+
+  if [ -z "$LATEST_RELEASE" ]; then
+
+    # generate only manifests served at /<version>
+    jekyll build --config _config.yml,$(pwd)/netlify/_config_noindex.yml,$(pwd)/netlify/_manifests_only.yml --baseurl /$RELEASE_VERSION --destination _site/$RELEASE_VERSION
+
+    # generate the website which is served at /archive/<version>/
+    # note: this must be done second since destination folders are cleaned on each builds
+    jekyll build --config _config.yml,$(pwd)/netlify/_config_noindex.yml --baseurl /archive/$RELEASE_VERSION --destination _site/archive/$RELEASE_VERSION
+
+    # move 404 page to root of the site so that netlify shows it automatically
+    mv _site/archive/$RELEASE_VERSION/404.html _site/404.html
+
+  else
```

Comment:
This is a pretty large conditional statement, and a lot of it seems the same in both paths. Can you trim it down by conditionally setting a variable which is then used in both places or something?

Reply:
An offline discussion revealed that sure,  we could build (`/` , `/version` , `/archive/version` ) every time, in that order, regardless of `LATEST_RELEASE`. It would eliminate the duplicate code, make it easier for future contributors to make adjustments to the code, and would eliminate the extra release step of setting env vars in the netlify UI.

However, we'd need to verify there aren't any unintended consequences. And it's not clear how we'd account for some differences in the `--config` flags, e.g. _config_latest.yml.